internal/elasticsearch/client.go

```diff
@@ -448,3 +196,90 @@ func extractSecret(secretName, namespace string, client client.Client) {
 		}
 	}
 }
+
+func oauthEsClient(esAddr, token, caPath, certPath, keyPath, clusterName, namespace string) (*elasticsearch6.Client, error) {
```

Comment:
so, i think we could have actually kept what we had before but slightly modify it for using the ES apis --

instead of returning a http client instead we would return the http.Transport object as a part of the function, and then we can create the ES client (passing in the http.Transport object) in a single location.

This way we would be creating the elasticsearch6.Client object in a single location rather than two...

Reply:
Thanks @ewolinetz. I pushed latest changes after incorporating your comments.